includes/admin/views/html-quick-edit-product.php

```diff
@@ -174,11 +175,15 @@
 			<span class="input-text-wrap">
 				<select class="stock_status" name="_stock_status">
 					<?php
+					echo '<option value="" id="stock_status_no_change">' . esc_html__( '— No Change —', 'woocommerce' ) . '</option>';
```

Comment:
This is done via JS (so that for variable products this one is selected, and for other products the old logic of selecting the current status is kept): https://github.com/woocommerce/woocommerce/pull/26174/files#diff-c3dfd13ac5cb3cdf7741fbad7cc433e6R64-R65 (`assets/js/admin/quick-edit.js` 64-65)